llvm/test/Transforms/InstCombine/phi.ll

```diff
@@ -350,6 +351,9 @@ define i1 @test11() {
 ; CHECK-NEXT:    [[D:%.*]] = call i1 @test11a()
 ; CHECK-NEXT:    br i1 [[D]], label [[ONE]], label [[END]]
 ; CHECK:       end:
+; CHECK-NEXT:    [[I:%.*]] = ptrtoint ptr [[A]] to i64
+; CHECK-NEXT:    [[G:%.*]] = inttoptr i64 [[I]] to ptr
+; CHECK-NEXT:    store i32 10, ptr [[G]], align 4
```

Comment:
I think we could fold this by arguing that any provenance `inttoptr` here may pick cannot be better than the provenance of `%a`, which is dereferenceable for the full store size.